dotCMS/src/main/java/com/dotcms/rest/BundleResource.java

```diff
@@ -218,4 +240,295 @@ public Response deleteEnvironmentPushHistory(@Context HttpServletRequest request
+
+        for (final String bundleId : deleteBundlesByIdentifierForm.getIdentifiers()) {
+
+            this.bundleAPI.deleteBundleAndDependencies(bundleId, initData.getUser());
+        }
+    }
+
+    /**
+     * Deletes bundles older than a date. (unsent are not going to be deleted)
+     * Note: the response will be notified by socket message
+     * @param request   {@link HttpServletRequest}
+     * @param response  {@link HttpServletResponse}
+     * @param olderThan {@link ISODateParam} an ISO date, should be before now to be valid
+     */
+    @DELETE
+    @Path("/olderthan/{olderThan}")
+    @Produces("application/json")
+    public Response deleteBundlesOlderThan(@Context   final HttpServletRequest request,
+                                       @Context   final HttpServletResponse response,
+                                       @PathParam("olderThan") final ISODateParam olderThan) {
+
+        if(olderThan.after(new Date())) {
+
+            throw new IllegalArgumentException("To avoid deleting bundles that publish in the future, the date can not be after the current date");
+        }
+
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredBackendUser(true)
+                .requiredFrontendUser(false)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(true)
+                .init();
+
+        final Locale locale    = LocaleUtil.getLocale(request);
+
+        Logger.info(this, "Deleting the bundles older than: " + olderThan
+                + " by the user: " + initData.getUser().getUserId());
+
+        final DotSubmitter dotSubmitter = DotConcurrentFactory
+                .getInstance().getSubmitter(BUNDLE_THREAD_POOL_SUBMITTER_NAME);
+        dotSubmitter.execute(() -> {
+
+            try {
+
+                final int bundleDeletedSize =
+                        this.bundleAPI.deleteBundleAndDependenciesOlderThan(olderThan, initData.getUser()).size();
+
+                this.sendSuccessDeleteBundleMessage(bundleDeletedSize, initData, locale);
+            } catch (Exception e) {
+
+                Logger.error(this.getClass(),
+                        "Exception on deleteBundlesOlderThan, couldn't delete bundles older than: " + olderThan +
+                                ", exception message: " + e.getMessage(), e);
+
+                this.sendErrorDeleteBundleMessage(initData, locale, e);
+            }
+        });
+
+        return Response.ok(new ResponseEntityView(
+                "Removing bundles in a separated process, the result of the operation will be notified")).build();
+    } // deleteBundlesOlderThan.
+
+    /**
+     * Deletes all failed and succeed bundles
+     * Note: the response will be notified by socket message
+     * @param request   {@link HttpServletRequest}
+     * @param response  {@link HttpServletResponse}
+     */
+    @DELETE
+    @Path("/all")
+    @Produces("application/json")
```

Comment:
This should not be needed. If you do `DELETE /bundles` should be implicit that you are deleting them all